### PR TITLE
Update actions: Use token based authentication

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Switch `az login` authentication from Service Principal mechanism to token based mechanism.

Ref: https://learn.microsoft.com/en-us/azure/container-registry/container-registry-repository-scoped-permissions

### CODE REVIEW

File: .github/workflows/version-history.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
     name: Update version history
     if: ${{ github.ref == 'refs/heads/main' || github.event.base_ref == 'refs/heads/main' }}
     runs-on: ubuntu-latest
-    environment: documentation
+    environment: publishing
```
Comment (chrmarti) [inline .github/workflows/version-history.yml:32]: Is the token associated with the `publishing` environment?

File: .github/workflows/push-again.yml [hunk 2]
```diff
@@ -54,12 +49,14 @@ jobs:
         STUB_REGISTRY: ${{ secrets.STUB_REGISTRY }}
         STUB_REGISTRY_BASE_PATH: ${{ secrets.STUB_REGISTRY_BASE_PATH }}
         SECONDARY_REGISTRY_BASE_PATH: ${{ secrets.SECONDARY_REGISTRY_BASE_PATH }}
+        TOKEN_NAME: ${{ secrets.TOKEN_NAME }}
+        PASSWORD: ${{ secrets.PASSWORD }}
```
Comment (chrmarti) [inline .github/workflows/push-again.yml:53]: Does the token have an expiration date?

**Review by chrmarti — APPROVED** [review]
LGTM!

Are we using the ACR to let MCR pull from it or something similar? Would GHCR work instead?

File: .github/workflows/push-again.yml (hunk 2 above)
Comment (samruddhikhandale) [reply .github/workflows/push-again.yml:53]: Yes, it's similar to authenticating with Service Principals (~one year)

Once we move to another tenant, we'd be able to use Managed Identities which doesn't need password rotations.

**Comment by chrmarti** [comment]
> > Would GHCR work instead?
> 
> Won't we need to push our images to GHCR for that to work? 🤔 I don't think we'd do that as the images would break.

I guess we can't use GHCR to push to MCR? That would avoid the need for the ACR credentials and the GitHub workflows can easily be given access to their repository's namespace in GHCR.

But this would be longer term. I agree with the approach taken here. 👍

**Comment by samruddhikhandale** [comment]
> > > Would GHCR work instead?
> > 
> > 
> > Won't we need to push our images to GHCR for that to work? 🤔 I don't think we'd do that as the images would break.
> 
> I guess we can't use GHCR to push to MCR? That would avoid the need for the ACR credentials and the GitHub workflows can easily be given access to their repository's namespace in GHCR.
> 
> But this would be longer term. I agree with the approach taken here. 👍

Ah, got it. When I get to researching the moving of tenants task, I can see if this is even possible. Thanks for the info here.

**Pull request merged**